File: src/main/java/client/effects/user/ReservedEffect.java
package client.effects.user;

import client.effects.AbstractEffect;
import client.effects.EffectType;
import util.packet.PacketWriter;

public class ReservedEffect extends AbstractEffect {

    private final String path;

    public ReservedEffect(String path) {
        this.path = path;
    }

    @Override
    public EffectType getType() {
        return EffectType.RESERVED_EFFECT;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeMapleString(path);
    }
}

File: src/main/java/client/effects/user/ReservedEffect.kt
package client.effects.user

import client.effects.AbstractEffect
import client.effects.EffectType
import util.packet.PacketWriter

class ReservedEffect(val path: String) : AbstractEffect() {

    override val type = EffectType.RESERVED_EFFECT

    override fun encodeData(pw: PacketWriter) {
        pw.writeMapleString(path)
    }
}